app/views/posts/index.html.erb

```diff
@@ -0,0 +1,27 @@
+<p id='notice'><%= notice %></p>
+
+<h1>Posts</h1>
+
+<table>
+  <thead>
+    <tr>
+      <th>Post</th>
+      <th colspan='3'></th>
+    </tr>
+  </thead>
+
+  <tbody>
+    <% @posts.each do |post| %>
+      <tr>
+        <% if user_signed_in? %>
+        <td><%=post.user.name%></td>
+        <%end%>
```

Comment:
We could define a helper method like this in posts_helper
```
  def post_author(post)
    if user_signed_in?
      post.user.name
    end
  end
```
And use the method to render the post author name within a td tag in our view.